gh-actions/go/generate/action.yaml

```diff
@@ -52,10 +52,21 @@ runs:
       with:
         token: ${{ inputs.token }}
     - name: Install latest protoc
-      uses: arduino/setup-protoc@v3
       if: ${{ steps.proto-deps.outputs.needs-protoc == 'true' }}
-      with:
-        repo-token: ${{ inputs.token }}
+      run: |
+        AUTH_HEADER=""
+        if [ -n "${{ inputs.token }}" ]; then
+          AUTH_HEADER="-H 'Authorization: Bearer ${{ inputs.token }}'"
+        fi
+        PROTOC_VERSION=$(eval curl -s $AUTH_HEADER \
+          https://api.github.com/repos/protocolbuffers/protobuf/releases/latest \
+          | jq -r '.tag_name' | sed 's/^v//')
+        curl -LO https://github.com/protocolbuffers/protobuf/releases/download/v${PROTOC_VERSION}/protoc-${PROTOC_VERSION}-linux-x86_64.zip
+        unzip -o protoc-${PROTOC_VERSION}-linux-x86_64.zip -d $HOME/.local
+        rm protoc-${PROTOC_VERSION}-linux-x86_64.zip
+        echo "$HOME/.local/bin" >> $GITHUB_PATH
+      shell: bash
```

Comment:
nitpick: I think in most actions, we pass the shell before `run`, just look at other actions and please align. If we are not consistent, my bad for letting that pass through :)

Reply:
That would be excellent! Hopefully, I was consistent in my own actions :)

I agree with a commit and PR to align all of this, include `set -eu` and `name of steps`!

Reply:
Saw it :)